readme.md

```diff
@@ -15,834 +14,1113 @@
+Only resort to aliasing when no preferable option is available.
 
-    // constructor function
-    PascalCase;
-    // but camel case when making public:
-    CKEDITOR.sth.camelCase;
+```javascript
+function Device( opts ) {
+	var self = this;
+	stream.read( opts.path, function( value ) {
+		self.value = value;
+	} );
+}
+```
 
+## 2.8. Misc
 
-    // 6.A.3.6
-    // From the Google Closure Library Style Guide
+This section will serve to illustrate ideas and concepts that should not be considered dogma, but instead exists to encourage questioning practices in an attempt to find better ways to do common JavaScript programming tasks.
 
-    functionNamesLikeThis;
-    variableNamesLikeThis;
-    ConstructorNamesLikeThis;
-    EnumNamesLikeThis;
-    methodNamesLikeThis;
-    SYMBOLIC_CONSTANTS_LIKE_THIS;
+### 2.8.1. Early returns
 
-    ```
+Early returns promote code readability with negligible performance difference.
+```javascript
+// Bad:
+function returnLate( foo ) {
+	var ret;
 
-    B. Faces of `this`
+	if ( foo ) {
+		ret = 'foo';
+	} else {
+		ret = 'quux';
+	}
+	return ret;
+}
 
-    Keep `this` "local" to the class. This means &ndash; do not bind e.g. a function in `selection.js` file to editor instance and do not create a function which should be ran in editor context. This way you'll avoid confusion what `this` refers to.
+// Good:
+function returnEarly( foo ) {
+	if ( foo ) {
+		return 'foo';
+	}
+	return 'quux';
+}
 
-    ```javascript
+// Bad:
+function doSthComplex( foo, bar ) {
+	if ( foo ) {
+		// do something
 
-    // 6.B.1
-    function Selection() {
-        var editor = this.editor;
+		if ( bar ) {
+			// do something
+		} else {
+			// do something else
+		}
+	}
+}
 
-        // Bad:
-        editor.on( 'setData', someCallback, editor );
+// Good:
+function doSthComplex( foo, bar ) {
+	if ( !foo )
+		return;
 
-        // Still bad:
-        editor.on( 'setData', CKEDITOR.tools.bind( someCallback, editor ) );
+	// do something
 
-        // Good:
-        editor.on( 'setData', someCallback, this );
+	if ( !bar ) {
+		// do something else
+		return;
+	}
 
-        // Good:
-        this.getEditor().on( 'setData', function( evt ) {
-            evt.editor.doSomething();
-        } );
+	// do something
+}
+```
 
-        // Good too:
-        var that = this;
-        editor.on( 'setData', function( evt ) {
-            editor.setCurrentSelection( that );
-        } );
-    }
+### 2.8.2. Objects minify poorly, closures very well
 
-    function someCallback() {
-        // Should refers to Selection, not Editor.
-        this;
-    }
+Avoid excessing usage of objects, when functional programming may be used as well. Properties and method names unlike private variables and function names cannot be shortened by minifier.
+You can play with [online Google Closure Compiler](http://closure-compiler.appspot.com/home) to test different approaches. Make sure to check the gzipped size which matters the most.
```

Comment:
I'm not so sure about considering only size. We should also consider execution time, which is not 100% bound to the size of downloadable content.

Reply:
> Avoid excessing usage of objects, when functional programming may be used as well

But here you are writing about implementation correctness, not style guide :thinking: